lib/twitter/direct_message_event.rb

```diff
@@ -23,11 +24,15 @@ def initialize(attrs)
 
   private
 
+    def read_from_response(attrs)
+      attrs[:event].nil? ? attrs : attrs[:event]
+    end
+
     def build_direct_message(attrs, text)
       recipient_id = attrs[:message_create][:target][:recipient_id].to_i
       sender_id = attrs[:message_create][:sender_id].to_i
       {id: attrs[:id].to_i,
-       created_at: Time.at(attrs[:created_timestamp].to_i / 1000.0),
+       created_at: Time.at(attrs[:created_timestamp].to_i / 1000.0).to_s,
```

Comment:
Curious why you made this change? I’d prefer times be represented as `Time` objects instead of `String`s.

Reply:
I was getting an error, I think on create_direct_message (but not totally sure -- would have to investigate) when I ran it from our app, and this fixed that.  I can try to dig a little deeper and see if there was a different place that I could fix it.

Reply:
It needs to be a string if you want to be able to use the `created_at` method on direct message.  `Time.parse` takes a string.  It can't handle a Time object.  The `created_at` method will return a Time object when given a string.

Reply:
I’d rather we fix that upstream. There’s no need to call `Time.parse` if we already have a `Time` object. Can you modify the `Twitter::Creatable` module not to parse if we already have a time? Otherwise, we’re converting from a JSON string to a `Time` object, back to a string, and then back to a `Time`.

Maybe something like this:

```ruby
def created_at
  time = @attrs[:created_at]
  Time.parse(time).utc unless time.nil? || time.is_a?(Time)
end
```

For this to work, we also must ensure that the `Time` is in UTC instead of local time:

```ruby
created_at: Time.at(attrs[:created_timestamp].to_i / 1000.0).utc
```

Reply:
Will fix, right now.

Reply:
To me, it made more sense to just ensure utc in the `created_at` method, since it sounds like we always want it to return a time object in utc.  I could have done it a couple ways, but I chose to follow what you wrote as closely as possible with that one modification.  Will be happy to put it into whatever form you prefer.